soc/st/stm32/stm32mp13x/soc.c

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for STM32MP13 processor
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/linker/linker-defs.h>
+
+#include <stm32_ll_bus.h>
+#include <cmsis_core.h>
+
+/**
+ * @brief Perform basic hardware initialization at boot.
+ *
+ * This needs to be run from the very beginning.
+ * So the init priority has to be 0 (zero).
+ *
+ * @return 0
+ */
+#define VECTOR_ADDRESS ((uintptr_t)_vector_start)
+
+void relocate_vector_table(void)
+{
+	write_sctlr(read_sctlr() & ~HIVECS);
+	write_vbar(VECTOR_ADDRESS & VBAR_MASK);
+	barrier_isync_fence_full();
+}
+
+void soc_early_init_hook(void)
+{
+	/* Update CMSIS SystemCoreClock variable (HCLK) */
+	SystemCoreClock = 1000000000U;
+
+	/* Clear TE bit to take exceptions in Thumb mode to fix the DDR init */
+	write_sctlr(read_sctlr() & ~SCTLR_TE_Msk);
+	barrier_isync_fence_full();
+}
+
+static const struct arm_mmu_region mmu_regions[] = {
+	MMU_REGION_FLAT_ENTRY("vectors", 0xC0000000, 0x1000, MPERM_R | MPERM_X | MT_NORMAL),
+
+	MMU_REGION_FLAT_ENTRY("GIC", 0xA0021000, 0x7000, MPERM_R | MPERM_W | MT_DEVICE),
+
+	MMU_REGION_FLAT_ENTRY("APB1", 0x40000000, 0x19400, MPERM_R | MPERM_W | MT_DEVICE),
+
+	MMU_REGION_FLAT_ENTRY("APB2", 0x44000000, 0x14000, MPERM_R | MPERM_W | MT_DEVICE),
+
+	MMU_REGION_FLAT_ENTRY("AHB2", 0x48000000, 0x40000, MPERM_R | MPERM_W | MT_DEVICE),
```

Comment:
last periph on the bus is OTG(0x49000000 - 0x4903FFFF)
```suggestion
	MMU_REGION_FLAT_ENTRY("AHB2", 0x48000000, 0x140000, MPERM_R | MPERM_W | MT_DEVICE),
```

Reply:
@arnopo is right: AHB2.
Regarding th address space, AHB2 include 2 areas:
0x4800000..0x48006400 (DMAs, ADCs)
0x4900000..0x49040000 (OTGs)

Reply:
Done